test/byContract/IexecPocoBoost/IexecPocoBoostDelegate.test.ts

```diff
@@ -391,6 +397,190 @@ describe('IexecPocoBoostDelegate', function () {
+            await expectOrderConsumed(iexecPocoBoostInstance, datasetOrderHash, undefined);
+            await expectOrderConsumed(iexecPocoBoostInstance, workerpoolOrderHash, undefined);
+            await expectOrderConsumed(iexecPocoBoostInstance, requestOrderHash, undefined);
+            const startTime = await setNextBlockTimestamp();
+            const matchOrdersArgs = [
+                appOrder,
+                datasetOrder,
+                workerpoolOrder,
+                requestOrder,
+            ] as Orders;
+
+            expect(
+                await iexecPocoBoostInstance
+                    .connect(sponsor)
+                    .callStatic.sponsorMatchOrdersBoost(...matchOrdersArgs),
+            ).to.equal(dealId);
+            await expect(
+                iexecPocoBoostInstance.connect(sponsor).sponsorMatchOrdersBoost(...matchOrdersArgs),
+            )
+                .to.emit(iexecPocoBoostInstance, 'SchedulerNoticeBoost')
+                .withArgs(
+                    workerpoolInstance.address,
+                    dealId,
+                    appInstance.address,
+                    datasetInstance.address,
+                    requestOrder.category,
+                    teeDealTag,
+                    requestOrder.params,
+                    beneficiary.address,
+                )
+                .to.emit(iexecPocoBoostInstance, 'OrdersMatched')
+                .withArgs(
+                    dealId,
+                    appOrderHash,
+                    datasetOrderHash,
+                    workerpoolOrderHash,
+                    requestOrderHash,
+                    expectedVolume,
+                )
+                .to.emit(iexecPocoBoostInstance, 'Transfer')
+                .withArgs(sponsor.address, iexecPocoBoostInstance.address, dealPrice)
+                .to.emit(iexecPocoBoostInstance, 'Lock')
+                .withArgs(sponsor.address, dealPrice)
+                .to.emit(iexecPocoBoostInstance, 'Transfer')
+                .withArgs(scheduler.address, iexecPocoBoostInstance.address, schedulerStake)
+                .to.emit(iexecPocoBoostInstance, 'Lock')
+                .withArgs(scheduler.address, schedulerStake)
+                .to.emit(iexecPocoBoostInstance, 'DealSponsoredBoost')
+                .withArgs(dealId, sponsor.address);
+            await expectOrderConsumed(iexecPocoBoostInstance, appOrderHash, expectedVolume);
+            await expectOrderConsumed(iexecPocoBoostInstance, datasetOrderHash, expectedVolume);
+            await expectOrderConsumed(iexecPocoBoostInstance, workerpoolOrderHash, expectedVolume);
+            await expectOrderConsumed(iexecPocoBoostInstance, requestOrderHash, expectedVolume);
+            const deal = await viewDealBoost(dealId);
+            // Check addresses.
+            expect(deal.requester).to.be.equal(requestOrder.requester, 'Requester mismatch');
```

Comment:
Imo you can remove error messages but you can keep it like that if you prefer

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/67/commits/c3bea95a2cbc2b9b10fdcd169b695da68ddbdeeb)